protocols/mdns/src/behaviour.rs

```diff
@@ -18,33 +18,80 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-use crate::service::{MdnsPacket, MdnsService, build_query_response, build_service_discovery_response};
-use async_io::Timer;
+use crate::dns::{build_query, build_query_response, build_service_discovery_response};
+use crate::query::MdnsPacket;
+use async_io::{Async, Timer};
 use futures::prelude::*;
+use if_watch::{IfEvent, IfWatcher};
+use lazy_static::lazy_static;
 use libp2p_core::{
-    Multiaddr,
-    PeerId,
-    address_translation,
-    connection::ConnectionId,
-    multiaddr::Protocol
+    address_translation, connection::ConnectionId, multiaddr::Protocol, Multiaddr, PeerId,
 };
 use libp2p_swarm::{
-    NetworkBehaviour,
-    NetworkBehaviourAction,
-    PollParameters,
-    ProtocolsHandler,
-    protocols_handler::DummyProtocolsHandler
+    protocols_handler::DummyProtocolsHandler, NetworkBehaviour, NetworkBehaviourAction,
+    PollParameters, ProtocolsHandler,
 };
 use smallvec::SmallVec;
-use std::{cmp, fmt, io, iter, mem, pin::Pin, time::{Duration, Instant}, task::Context, task::Poll};
+use socket2::{Domain, Socket, Type};
+use std::{
+    cmp,
+    collections::VecDeque,
+    fmt, io, iter,
+    net::{IpAddr, Ipv4Addr, SocketAddr, UdpSocket},
+    pin::Pin,
+    task::Context,
+    task::Poll,
+    time::{Duration, Instant},
+};
 
-const MDNS_RESPONSE_TTL: std::time::Duration = Duration::from_secs(5 * 60);
+lazy_static! {
+    static ref IPV4_MDNS_MULTICAST_ADDRESS: SocketAddr =
+        SocketAddr::from((Ipv4Addr::new(224, 0, 0, 251), 5353));
+}
+
+pub struct MdnsConfig {
+    /// TTL to use for mdns records.
+    pub ttl: Duration,
+    /// Interval at which to poll the network for new peers. This isn't
+    /// necessary during normal operation but avoids the case that an
+    /// initial packet was lost and not discovering any peers until a new
+    /// peer joins the network. Receiving an mdns packet resets the timer
+    /// preventing unnecessary traffic.
+    pub interval: Duration,
```

Comment:
```suggestion
    pub query_interval: Duration,
```

Would be more descriptive in my eyes.